packages/core/src/lib/social.ts

```diff
@@ -88,15 +83,23 @@ export const getUserInfoFromInteractionResult = async (
 export const findSocialRelatedUser = async (
   info: SocialUserInfo
 ): Promise<Nullable<[{ type: 'email' | 'phone'; value: string }, User]>> => {
-  if (info.phone && (await hasUserWithPhone(info.phone))) {
+  if (info.phone) {
     const user = await findUserByPhone(info.phone);
 
+    if (!user) {
```

Comment:
what if the social returns both phone and email,  and we have an email in the record but not the phone?

Reply:
my fault, updated